### PR TITLE
Fix errors and warnings reported by Jekyll 3.5.0

### PR DESCRIPTION
**Pull request by ptsurbeleu**
Hey Ryan,

Here is the version of your blog with all the fixes for issues & warnings reported by Jekyll 3.5.0. I also have an online version of these changes running just fine on Jekyll 3.5.0: http://06701527.pabloduo.com.

Let me know what you think and/or if you spot something is wrong or missing.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 source 'https://rubygems.org'
-gem 'jekyll', '2.5.3'
+gem 'jekyll', '3.5.0'
```
Comment (ptsurbeleu) [inline Gemfile:2]: This is an actual update to run with v3.5.0

File: _config.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ name: 'Ryan Straits'
 author: 'Ryan Straits'
 author_image: "/assets/images/RMS-large-cntr.jpg"
 
+# plugins
+plugins: ['jekyll-paginate']
```
Comment (ptsurbeleu) [inline _config.yml:21]: Starting from v3.x.x you must also declare the list of Jekyll plugins to be loaded. It is used to be 'gems' and v3.5.0 has renamed it to 'plugins'.

File: _layouts/post.html [hunk 3]
```diff
@@ -81,14 +81,14 @@ <h5 class="index-headline featured"><span>Written by</span></h5>
               <p class="published">Published <time datetime="{{ page.date | date: '%F %R' }}">{{ page.date | date_to_string }}</time></p>
             </section>
           </div>
-          {{/post}}
+          {{post}}
```
Comment (ptsurbeleu) [inline _layouts/post.html:84]: It was a warning reported by Jekyll and it looks like an accidental typo to me...

File: _layouts/post.html [hunk 4]
```diff
@@ -81,14 +81,14 @@ <h5 class="index-headline featured"><span>Written by</span></h5>
...
           <div class="isRight">
             <h5 class="index-headline featured"><span>Supported by</span></h5>
             <footer class="site-footer">
               <section class="poweredby">Proudly published with <a href="http://jekyllrb.com"> Jekyll</a></section>
               <a class="subscribe" href="{{ "/feed.xml" | prepend: site.baseurl }}"> <span class="tooltip"> <i class="fa fa-rss"></i> You should subscribe to my feed.</span></a>
               <div class="inner">
-                <section class="copyright">All content copyright <a href="{{@blog.url}}/">{{site.author}}</a> &copy; {{ site.time | date: '%Y' }}<br>All rights reserved.</section>
+                <section class="copyright">All content copyright <a href="{{blog.url}}/">{{site.author}}</a> &copy; {{ site.time | date: '%Y' }}<br>All rights reserved.</section>
```
Comment (ptsurbeleu) [inline _layouts/post.html:91]: Another thing reported as a warning by Jekyll and I'm actually not sure about that one... Maybe you can help me to get more context on this one.

File: _posts/2003-01-29-008.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ tags:
 - xanga
 comments: true
 image: "/assets/images/xanga/2004-04-16.jpg"
-slug: 008
+slug: '008'
```
Comment (ptsurbeleu) [inline _posts/2003-01-29-008.md:11]: These ones are epic! And there are lots of those! Without quotes these slugs were treated by YAML as numbers and not strings!!! Which lead to weird error messages and without debugging it with `pry` I would not be able to figure out what the heck was going on...

File: _layouts/post.html (hunk 4 above)
Comment (ptsurbeleu) [inline _layouts/post.html:91]: I'm wondering if `{{blog.url}}` is a typo as well and should be `{{site.url}}` instead? I have looked thru [Variables](https://jekyllrb.com/docs/variables/) section in Jekyll's documentation and couldn't find anything even closely related to an object named `blog`...

Comment (ryanstraits) [reply _layouts/post.html:91]: I believe that came with the theme that I copied & modified. I didn't get deep enough to change variables, I believe. Thank you so much for your help. I'm going to merge this pull.

**Comment by ryanstraits** [comment]
Got this error upon page build: 

"The page build completed successfully, but returned the following warning for the `master` branch:

You are currently using the 'redcarpet' Markdown engine, which is no longer supported by GitHub Pages and may cease working at any time. To ensure your site continues to build, remove the 'markdown' setting in your site's '_config.yml' file and confirm your site renders as expected. For more information, see https://help.github.com/articles/updating-your-markdown-processor-to-kramdown/."

**Comment by ptsurbeleu** [comment]
Hmm, this one is weird but I believe you should be able to trust Jekyll and simply remove that setting from `_config.yml`... Lets see what happens 😃

File: _layouts/post.html (hunk 4 above)
Comment (ptsurbeleu) [reply _layouts/post.html:91]: Not a problem at all! It was a lot of fun afterall to give it a try 😃

**Comment by ryanstraits** [comment]
Hah ok I'll give it a shot.

**Comment by ptsurbeleu** [comment]
Did it work?

**Comment by ryanstraits** [comment]
Yes, it did! Currently going to try working on a new theme. That's why I wanted to get these issues figured out first. Really appreciate your help! Best regards, Ryan

**Comment by ptsurbeleu** [comment]
Yay! It is awesome! Always welcome! :smiley:

Can I ask you about a favor also update the original thread in Jekyll Talk?

**Comment by ryanstraits** [comment]
Sure, what would you like me to update it with? Is there a way to close it?

**Comment by ptsurbeleu** [comment]
It would be great if you can share links to the PR commit and others that you did to address warnings & stuff.

I saw a few other threads where folks were asking for advice how to migrate to the latest Jekyll and your experience might server as a foundation for other cases.

Thanks a ton!

**Comment by ptsurbeleu** [comment]
I think I have seen some Help threads went into Solved or Closed state.

**Pull request merged**